src/Microsoft.AspNet.Razor/CodeGenerators/CSharpLineMappingWriter.cs

```diff
@@ -114,7 +114,7 @@ public void Dispose()
             {
                 // Need to add an additional line at the end IF there wasn't one already written.
                 // This is needed to work with the C# editor's handling of #line ...
-                var endsWithNewline = _writer.GenerateCode().EndsWith("\n");
```

Comment:
This seems to be buggy on CoreCLR \ Linux. https://github.com/dotnet/corefx/issues/4587